src/goDebugConfiguration.ts

```diff
@@ -29,16 +29,22 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
 			if (!activeEditor || activeEditor.document.languageId !== 'go') {
 				return;
 			}
+			let debugMode;
+			let testFileRegExp = RegExp('.*_test\.go$', 'g');
+			if (testFileRegExp.test(activeEditor.document.fileName) === true) {
+				debugMode = 'test';
+			} else {
+				debugMode = 'debug';
+			}
 
 			debugConfiguration = {
 				'name': 'Launch',
 				'type': 'go',
 				'request': 'launch',
-				'mode': 'debug',
+				'mode': debugMode,

```

Comment:
When there is a debug configuration in the launch.json file, this `if` block (starting at line 27) will not be entered. So this fix will only get applied to cases when there is no debug configuration in the launch.json file.

I would suggest to update the mode after this `if` block